draft-smyslov-ipsecme-ikev2-mceliece-00.xml

```diff
@@ -101,23 +103,41 @@
             group offered in an ADDKE payload if the initiator has not negotiated the support for both big
             payloads and for fragmentation.</t>
           <t> Within the IKE_INTERMEDIATE exchange, such payloads will normally encode the large public
-            keys. A diagram is provided here for illustration:</t>
+            keys. A diagram is provided here for illustration:. The format below is the big payload version
+            of the Key Exchange from section 3.4 of <xref target="RFC7296"/>.</t>
          <figure> 
             <artwork><![CDATA[
                         1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
    | Next Payload  |C|L| RESERVED  |         Payload Length        
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      continued.                   | The public key of the McEliece
+      length continued...          |   Diffie-Hellman Group Num    | 
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+   
-    KEM begins here...
+   |           RESERVED            | McEliece KEM key data         |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               +
+   |                                                               |
+   ~                   KEM key data continues...                   ~
+   |                                                               |

```

Comment:
I also wonder whether we should use more neutral text here, since iy is either public key or ciphertext (which is not a key). F related issue is that we should state that big payloads are only necessary for transferring McEliece public key, since the cize of its ciphertext is small. But I think we should not prohibit to use big payload for ciphertext too (it can simplify implementation). Thus something like "MAY" keyword for use of big payloads for ciphertext...